v2/api/v2/egress_types.go

```diff
@@ -59,6 +66,10 @@ type EgressSpec struct {
 	// The default is false.
 	// +optional
 	FouSourcePortAuto bool `json:"fouSourcePortAuto,omitempty"`
+
+	// PodDisruptionBudget is an optional PodDisruptionBudget for Egress NAT pods.
+	// +optional
+	PodDisruptionBudget *policyv1.PodDisruptionBudgetSpec `json:"podDisruptionBudget,omitempty"`
```

Comment:
If we only support `maxUnavailable` and `minAvailable`, and we do not plan to support the other fields in the PDB, I think we should use our own struct.
When using `policyv1.PodDisruptionBudgetSpec`, `kubectl explain egress.spec.podDisruptionBudget` shows original PDB's explanations and that will confuse users.

Reply:
To my opinion, using `policyv1.PodDisruptionBudgetSpec` directly is a good solution.
For this kind of option there would be specific requirements for each user, so we would get individual PRs to match their requirements that just copy-paste the fields of `PodDisruptionBudgetSpec`.

Reply:
What exactly does "specific requirements" mean?
Is it something that can only be satisfied with using `policyv1.PodDisruptionBudgetSpec`?

Reply:
> but if someone wants to use unhealthyPodEvictionPolicy field, they should create a PR to use it and we should approve it. 

I agree. And therefore, I don't think we should use `policyv1.PodDisruptionBudgetSpec` because, we don't support `unhealthyPodEvictionPolicy` (and some other fields).
I think we should set only fields that we support.

I wonder what is the advantage of using `policyv1.PodDisruptionBudgetSpec`.
If we just want to be able to set values from existing PDB's spec, I think it's enough that just set the field name to same as PDB as follows.
```go
type EgressPodDisruptionBudget struct {
	MinAvailable *intstr.IntOrString `json:"minAvailable"`
	MaxUnavailable *intstr.IntOrString `json:"maxUnavailable"`
}
```

Reply:
> Also, PDB spec may be expanded in future versions and it needs extra burden to support fields one-by-one.

Sorry, I missed this line.

I don't think there isn't much burden whether we use `policyv1.PodDisruptionBudgetSpec` or not.
The difference in implementation when the new field is added is that add a field to original struct I think.
I think it would be more troubling that the fields are added silently when the dependent package is updated.

Reply:
The burden is not only on the authors' side but also on the users' side.
If you are a user of an OSS and want to use some specific field of a upstream struct (e.g. PodDisruptionBudgetSpec) that is not supported, you first need to make negotiation with the software authors, make a PR, wait for a release and then finally you can use it. Isn't it a huge amount of burden?

I suspect we can choose to delegate to users the responsibility to use the software safely, if it make easier for users to fit to their demands.

Reply:
We decided to use our own struct instead of `policyv1.PodDisruptionBudgetSpec` and support only `minAvailable` and `maxUnavailable` in terms of maintenance costs and user experience provided by validation.